sql/src/main/java/io/crate/action/sql/SQLOperations.java

```diff
@@ -280,7 +284,10 @@ public void execute(String portalName, int maxRows, ResultReceiver resultReceive
                     clearState();
                     ListenableFuture<?> result = portal.sync(planner, statsTables);
                     if (UNNAMED.equals(portal.name())) {
-                        portal.close();
+                        // The Postgres Wire Protocol states that unnamed portals are implicitly closed.
+                        // However if we would close the portal here, we would also interrupt the result receiving operations.
+                        // Instead we mark it as finished instead and close it with the next bind() call if necessary()
+                        portal.finish();
```

Comment:
I think you are right, because even if `sync()` is called before `bind()` in case of server-side prepared statements, the `portal.close()` call in the `bind()` method would only abort result collection if there's something ongoing, which must not be the case anyway, right?